src/input.rs

```diff
@@ -16,83 +21,80 @@ pub(crate) struct App {
 }
 
 impl App {
-    fn stdin_replace(&self, is_tty: bool) -> Result<()> {
-        let mut buffer = Vec::with_capacity(256);
-        let stdin = std::io::stdin();
-        let mut handle = stdin.lock();
-        handle.read_to_end(&mut buffer)?;
-
-        let stdout = std::io::stdout();
-        let mut handle = stdout.lock();
-
-        handle.write_all(&if is_tty {
-            self.replacer.replace_preview(&buffer)
-        } else {
-            self.replacer.replace(&buffer)
-        })?;
-
-        Ok(())
-    }
-
     pub(crate) fn new(source: Source, replacer: Replacer) -> Self {
         Self { source, replacer }
     }
+
     pub(crate) fn run(&self, preview: bool) -> Result<()> {
-        let is_tty = std::io::stdout().is_terminal();
+        let sources: Vec<(PathBuf, Mmap)> = match &self.source {
+            Source::Stdin => {
+				let mut handle = stdin().lock();
+				let mut buf = Vec::new();
+				handle.read_to_end(&mut buf)?;
+				let mut mmap = MmapOptions::new()
+					.len(buf.len())
+					.map_anon()?;
+				mmap.copy_from_slice(&buf);
+				let mmap = mmap.make_read_only()?;
+				vec![(PathBuf::from("STDIN"), mmap)]
+			},
+            Source::Files(paths) => {
+				let mut refs = Vec::new();
+                for path in paths {
+                    if !path.exists() {
+                        return Err(Error::InvalidPath(path.clone()));
+                    }
+					let mmap = unsafe { Mmap::map(&File::open(path)?)? };
+                    refs.push((path.clone(), mmap));
+                }
+                refs
+            },
+        };
+        let needs_separator = sources.len() > 1;
 
-        match (&self.source, preview) {
-            (Source::Stdin, true) => self.stdin_replace(is_tty),
-            (Source::Stdin, false) => self.stdin_replace(is_tty),
-            (Source::Files(paths), false) => {
-                use rayon::prelude::*;
+        let replaced: Vec<_> = {
+            use rayon::prelude::*;
+            sources.par_iter()
+                .map(|(path, mmap)| {
+					let replaced = self.replacer.replace(mmap); 
```

Comment:
Its looking like real tabs are sneaking their way in :scream:. Running `cargo fmt` should fix that (and there's plenty of ways to do that automatically like editor settings, git hooks, etc.)

Reply:
Indeed, my nvim setup somehow didn't recognize indentation and adjust accordingly, although there's a plugin advertised to exactly doing that. Now I'm using VSC anyway ;)